ludwig/hyperopt/run.py

```diff
@@ -223,6 +224,11 @@ def hyperopt(
 
     hyperopt_config = full_config[HYPEROPT]
 
+    # Convert hyperopt config to hyperopt schema to populate with schema defaults
+    # This fills in missing splits, executor config, search_alg, etc.
+    hyperopt_config_from_schema = HyperoptConfig.Schema().load(hyperopt_config)
+    hyperopt_config = hyperopt_config_from_schema.to_dict()
```

Comment:
Would the ModelConfig class, e.g. `model_config.py::ModelConfig::_set_hyperopt_defaults()` be a more appropriate place for this code?

Reply:
That's reasonable, I can move it inside there.